thriftbp/client_middlewares.go

```diff
@@ -230,6 +230,10 @@ func SetDeadlineBudget(next thrift.TClient) thrift.TClient {
 				}
 				value := strconv.FormatInt(ms, 10)
 				ctx = thrift.SetHeader(ctx, HeaderDeadlineBudget, value)
+
+				headers := thrift.GetWriteHeaderList(ctx)
+				headers = append(headers, HeaderDeadlineBudget)
+				ctx = thrift.SetWriteHeaderList(ctx, headers)
```

Comment:
Should we have a helper method for this?

Reply:
this can be written in one line if you really wanted to:

```go
ctx = thrift.SetWriteHeaderList(ctx, append(thrift.GetWriteHeaderList(ctx), HeaderDeadlineBudget))
```

so... maybe?

Reply:
Less about the exact lines of code it takes and more just that there's this exact series of steps you have to follow when you want to do this, it's easy to forget to do this, having a helper might help with that?

Reply:
good point. done.